echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+    )
+
+    # ---- Extract bin length values (i.e. mid)
+    length_data_binned[ 'length_bin_value' ] = (
+        length_data_binned[ 'length_bin' ].apply( lambda x: x.mid )
+    )
+
+    # ---- Merge `length_data_filtered ` and `length_weight_sex`
+    sexed_unaged_model_weights = length_data_binned.merge( length_weight_sex ,
+                                                           on = [ 'length_bin' , 'sex' ] ,
+                                                           how = 'left' )
+    
+    # ---- Interpolate weights over length values for each sex (weighted by `length_count`)
+    interpolated_weights = (
+        sexed_unaged_model_weights
         .groupby( [ 'stratum_num' , 'sex' ] ) 
-        .apply( lambda df: pd.Series( { 'summed_haul_weights': df.weight_interp.sum( ) } ) ) 
-        .reset_index( ) 
-    ) 
-  
-    ### Normalize haul weights (Station 1) 
-    haul_sex_weights_normalized = ( 
-        haul_weights 
-        .merge( weight_strata_station.loc[ lambda x: x.station == 1 ] ,  
-                on = 'stratum_num' ) 
-        .assign( weight_normalized_station_1 = lambda df: ( 
-            df 
-            .groupby( [ 'stratum_num' ] ) 
-            .apply( lambda strata: strata[ 'stratum_weight' ] * strata[ 'summed_haul_weights' ] / 
-                                    ( df[ ( df.stratum_num == strata.name ) & ( df.sex == 'male' ) ][ 'summed_haul_weights' ].iloc[ 0 ] + 
-                                      df[ ( df.stratum_num == strata.name ) & ( df.sex == 'female' ) ][ 'summed_haul_weights' ].iloc[ 0 ] ) ) 
-            .reset_index( drop = True ) ) ) 
-        .merge( weight_strata , on = [ 'stratum_num' ] ) 
-        .assign( proportion_normalized_station_1 = lambda df: ( 
-            df.weight_normalized_station_1 / df.weight_stratum_total 
-        ) ) 
-        .assign( sex_proportion = lambda df: ( 
-            df 
-            .groupby( [ 'stratum_num' ] ) 
-            .apply( lambda strata: strata[ 'proportion_normalized_station_1' ] / 
-                                    ( df[ ( df.stratum_num == strata.name ) & ( df.sex == 'male' ) ][ 'proportion_normalized_station_1' ].iloc[ 0 ] + 
-                                      df[ ( df.stratum_num == strata.name ) & ( df.sex == 'female' ) ][ 'proportion_normalized_station_1' ].iloc[ 0 ] ) ) 
-            .reset_index( drop = True ) ) ) 
-        .merge( aged_proportions.loc[ : , [ 'stratum_num' , 'proportion_unaged_total' ] ] ,  
-                on = [ 'stratum_num' ] )         
-        .assign( proportion_unaged_sex = lambda x: x.sex_proportion * x.proportion_unaged_total ) 
-    ) 
+        .apply( lambda df: pd.Series( {
+                'weight_interp': ( np.interp( df.length ,
+                                              df.length_bin_value ,
+                                              df.weight_modeled ) * df.length_count ).sum( )
+            } ) )
```

Comment:
Let's not use interpolation but use the regression relationship to compute the estimated weight?